File: halcyon-system/src/main/java/com/marshal/halcyon/hr/controller/HrUnitController.java
package com.marshal.halcyon.hr.controller;

import com.marshal.halcyon.core.entity.ResponseData;
import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.hr.entity.HrUnit;
import com.marshal.halcyon.hr.service.HrUnitService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * @auth: Marshal
 * @date: 2019/1/9
 * @desc:
 */
@RestController
@RequestMapping("/hr/unit")
public class HrUnitController extends BaseController {

    @Autowired
    HrUnitService hrUnitService;

    /**
     * 部门信息查询
     *
     * @param condition
     * @param pageNum
     * @param pageSize
     * @return
     */
    @RequestMapping("/query")
    public ResponseData query(@RequestBody HrUnit condition, int pageNum, int pageSize) {
        List<HrUnit> list = hrUnitService.select(condition, pageNum, pageSize);
        return new ResponseData(list);
    }

    /**
     * 部门信息提交
     *
     * @param hrUnit
     * @return
     */
    @RequestMapping("/submit")
    public ResponseData submit(@RequestBody HrUnit hrUnit) {
        if (!getValidator().isValid(hrUnit)) {
            return new ResponseData(false, getValidator().getErrors(hrUnit));
        }
        hrUnitService.save(hrUnit);
        return new ResponseData(true, "保存成功");
    }

    /**
     * 部门信息删除
     *
     * @param selectedIds
     * @return
     */
    @RequestMapping("/remove")
    public ResponseData remove(@RequestParam("selectedIds") Long[] selectedIds) {
        hrUnitService.batchDelete(selectedIds);
        return new ResponseData(true, "删除成功");
    }

    /**
     * 根据部门id查询部门
     *
     * @param unitId
     * @return
     */
    @RequestMapping("/selectByUnitId")
    public HrUnit selectByUnitId(@RequestParam Long unitId) {
        return hrUnitService.selectByPrimaryKey(unitId);
    }

    @RequestMapping("/getParentUnitOptions")
    public List<Map> getOptions() {
        return hrUnitService.getOptions();
    }
}

File: halcyon-web/src/main/resources/static/js/controller/hr/hrUnitController.js
//控制层
app.controller("hrUnitController", function ($scope, $controller, hrUnitService, hrPositionService) {
    //继承
    $controller("baseController", {$scope: $scope});

    //初始化
    $scope.initForAdd = function () {
        $scope.hrUnit = {};
        $scope.getParentUnitOptions();
    }
    $scope.initForUpdate = function (id) {
        hrUnitService.queryById(id).success(function (data) {
            $scope.hrUnit = data;
        });
        $scope.getParentUnitOptions();
        $scope.getManagerPositionOptions(id);
    }

    //crud
    //查询,初始化查询条件为空
    $scope.condition = {};
    $scope.query = function (pageNum, pageSize) {
        hrUnitService.query(pageNum, pageSize, $scope.condition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //保存
    $scope.save = function () {
        hrUnitService.save($scope.hrUnit).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //删除
    $scope.delete = function () {
        $scope.deleteConfirm(deleteRows);
    }

    function deleteRows() {
        hrUnitService.delete($scope.selectedIds).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }

    //获取下拉框数据
    //防止ajax异步导致为空，声明一个空的
    $scope.parentUnitOptions = {data: []};
    $scope.getParentUnitOptions = function () {
        hrUnitService.getParentUnitOptions().success(function (data) {
            $scope.parentUnitOptions = {data: data};
        });
    }

    $scope.managerPositionOptions = {data: []};
    $scope.getManagerPositionOptions = function (unitId) {
        hrPositionService.selectByUnitId(unitId).success(function (data) {
            $scope.managerPositionOptions = {data: data};
        });
    }
});